docs/frontend/in-depth/autosave.md

```diff
@@ -0,0 +1,154 @@
+        }
+        ```
+        - When fieldKeys are defined
+        ```javascript
+        {
+            createAttributeModalForm: {
+                myCustomFieldName: "Type of attribute is required"
+            }
+        }
+        ```
+<div class="Alert Alert--info">
+    Flattening supports nested structures.
+</div>
+
+### Saving data per context at which user have focus
+---
+
+The whole idea is to save data per current visible context. Each tab / form has his own save button which will save only data at given context. 
+
+<div class="Alert Alert--warning">
+    Currently it’s mixed and will be changed when BE application will adapt required changes.
+</div>
+
+### Giving feedback
+---
+
+There are 3 levels of giving displaying user feedback at tabs:
+- Changes have been made will be acquired by passing **`changeValues`** props to **`HorizontalRoutingTabBar`** component. The passed object should have structure of:
+
+    ```javascript
+    {
+        productGeneralTab: {
+            categories: ["abc68d44-732b-47df-8398-7c8038dc1133"],
+            saved: false,
+        },
+        productTemplateTab: {
+            "en_GB|text123/en_GB": "123", // As we can see the field key might be totally customased.
+            saved: false,
+        }  
+    }
+    ```
+
+        *Visual result*
+        <img src="images/tab_changes.png" alt="User made change at the tab indicator">
+
+- Presenting validation errors will be acquired by passing **`errors`** props to **`HorizontalRoutingTabBar`** component. The passed object should have structure of:
+
+    ```javascript
+    {
+        createProductModalForm: {
+            type: "Type is required",
+        }
+    }
+    ```
+
+    *Visual result*
+    
+    <img src="images/tab_errors.png" alt="User got errors after validating data at tab indicator">
+
+- Changes have been saved are achieved by invoking **`markChangeValuesAsSaved`** with **`scope`** of component (from the example above it will be `createProductModalForm`) from **`feedback`** store after successful request to BE.
+
+    <img src="images/tab_success.png" alt="User saved changes tab indicator">
+
+Additionally, we do support editing at Grid:
+
+- Changes have been made and not saved are handled by **`drafts`** props passed to **`Grid`** component. Structure of drafts should be written as:
+
+    ```javascript
+    {
+        drafts: {
+            "5ec31270-e518-4fdf-8e01-e20bfd48ca64/test123:en_GB": "0beb4ddb-2164-4b6e-aff8-0083c4ab4ecc",
+        }
+    }
+    ```
+  
+    Where:
+    - `5ec31270-e518-4fdf-8e01-e20bfd48ca64` is a row id
+    - `test123:en_GB` is a column id
+    - `0beb4ddb-2164-4b6e-aff8-0083c4ab4ecc` changed value of cell at given row and column
+    
+        *Visual result*
+        <img src="images/cell_success.png" alt="User made changes at grid">
+
+- Presenting validation errors will be acquired by passing **`errors`** props to **`Grid`** component. The passed object should have structure of:
```

Comment:
I think that we should create a Grid page and link it here. Grid component may be a placeholder page - will be completed later.

Reply:
I would leave it as it's now. Grid will be described in storybook.